BTrees/_base.py

```diff
@@ -815,9 +815,7 @@ def __delitem__(self, key):
         self._del(self._to_key(key))
 
     def clear(self):
-        if self._data:
-            # In the case of __init__, this was already set by __new__
-            self._data = []
+        self._data = []
```

Comment:
What about `if not hasattr(self, '_data') or self._data:` instead of the unconditional set, and extending the comment to mention that `_data` can sometimes be blown away entirely when the object is ghosted?  That seems more explicit to me, and has the minor benefit of avoiding object churn.